src/hotspot/share/ci/ciInstanceKlass.hpp

```diff
@@ -44,6 +44,7 @@ class ciInstanceKlass : public ciKlass {
   friend class ciMethod;
   friend class ciField;
   friend class ciReplay;
+  friend class CompileTrainingData;
```

Comment:
Not referenced here

Reply:
It allows `CompileTrainingData` to peek into the `ciInstanceKlass` internals. We need the klass ptr specially.

Reply:
I missed that it is "friend" declaration.